packages/application-extension/src/index.ts

```diff
@@ -264,17 +272,25 @@ const pages: JupyterFrontEndPlugin<void> = {
       }
     });
 
-    app.commands.addCommand(CommandIDs.openTree, {
-      label: trans.__('Open Files'),
-      execute: () => {
-        window.open(`${baseUrl}tree`);
-      }
-    });
+    if (!app.commands.isVisible('filebrowser:toggle-main')) {
```

Comment:
I was more wondering whether it would be possible to define `isVisible` as part of the command definition below.

Reply:
@brichet would you mind posting a summary of what the intended behavior should be, to make sure I understand correctly? Thanks!

Reply:
I may be confused about that, I'll try to clarify.

There are 2 different things:

1- What I was expected at first was a command to activate the file browser in tree view from another tab. For example nbgrader brings some tabs with links which open a path in file browser, using *filebrowser:go-to-path* command.  This command already exists in tree view, it is created when importing the plugin '@jupyterlab/filebrowser-extension:browser'. But at the end, this command call the command *filebrowser:activate*, which is not registered in Notebook yet. The result is that the path is opened but the file browser tab is not activated, so it look like nothing happened for user. 
This is fixed with the command below.

2- The plugin '@jupyterlab/filebrowser-extension:browser' also bring a command *filebrowser:toggle-main*, with a menu entry called `Open Files`, which call *filebrowser:activate* or *filebrowser:hide-main* depending on the file browser status. I thought we could use this entry already existing in tree view to open the file browser tab, instead of creating a new one which may be confusing.

Maybe the good thing to do is:
- modifying the creation of the command *filebrowser:toggle-main* in JupyterLab, to register it (and create menu entry) in the plugin '@jupyterlab/filebrowser-extension:browserWidget' instead of '@jupyterlab/filebrowser-extension:browser'. this would avoid importing it in Notebook and having this menu entry by default in tree view.
- creating a menu entry to open the file browser tab in Notebook when we create the command 'filebrowser:activate'.
- creating the menu entry to open a new `tree view` page only if the previous one does not exists (as you said).

Hope it clarifies.